contrib/babelfishpg_tsql/src/fts_parser.y

```diff
@@ -43,68 +45,100 @@ static void     replaceMultipleSpacesAndSpecialChars(char* input, char **str1, c
 %%
 
 contains_search_condition:
-    generation_term
-    | simple_term
-    | prefix_term
+    multiple_term {
+        *result = $1;
+    }
     ;
 
-simple_term:
-    WORD_TOKEN  {
-        *result = translate_simple_term($1);
+multiple_term:
+    search_term opt_node {
+        if($2 != NULL) 
+        {
+            $$ = mergeTokens($1, $2, NULL);
+        }
+        else
+        {
+            $$ = $1;
+        }
     }
-    | TEXT_TOKEN {
-        *result = translate_simple_term($1);
+    | enclosed_term opt_node {
+        if($2 != NULL) 
+        {
+            $$ = mergeTokens($1, $2, NULL);
+        }
+        else
+        {
+            $$ = $1;
+        }
     }
-    | WS_TOKEN WORD_TOKEN {
-        *result = translate_simple_term($2);
+    ;
+
+opt_node:
+    bool_operator multiple_term {
+        $$ = mergeTokens($1, $2, NULL);
     }
-    | WORD_TOKEN WS_TOKEN {
-        *result = translate_simple_term($1);
+    | /*EMPTY*/ {
+        $$ = NULL;
     }
-    | WS_TOKEN WORD_TOKEN WS_TOKEN {
-        *result = translate_simple_term($2);
+    ;
+
+enclosed_term:
+    white_space O_PAREN_TOKEN multiple_term C_PAREN_TOKEN white_space {
```

Comment:
Please check if multiple whitespaces are allowed or not. If allowed then please use `(WS_TOKEN)*` directly here. Please check at all the place where we are using such nodes.